app/src/components/template/Dashboard.js

```diff
@@ -11,22 +11,37 @@ import {
 import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
 import MenuIcon from '@material-ui/icons/Menu';
 import { withStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
 import { mainListItems } from './listItems';
 import { Styles as OBStyles } from '../../styles/OrderBookSnapshot';
 import OrderBookSnapshot from '../OrderBookSnapshot';
 import Home from '../Home';
 import NotFound from '../NotFound';
 import { styles } from '../../styles/Dashboard';
+import OrderBookService from '../../services/OrderBookService';
 
 class Dashboard extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
             open: false,
+            selectedInstrument: '',
+            instruments: [],
         };
     }
 
+    componentDidMount() {
+        const { instruments } = this.state;
+        const newInstruments = instruments.slice();
+        OrderBookService.getInstrumentsList().then(response => {
+            response.map(value => {
+                newInstruments.push(value);
+            });
+        });
+        this.setState({ instruments: newInstruments });
```

Comment:
asynchronousity is wrong here, the `setState` should happen in the `.then()` callback of your service call.

Reply:
👍